lib/routes/account.js

```diff
@@ -1179,19 +1179,20 @@ module.exports = function (
                */
               return db.verifyTokens(code, account)
                 .catch(function (err) {
-                  if (err.errno === error.ERRNO.INVALID_VERIFICATION_CODE && butil.buffersAreEqual(request.payload.code, account.emailCode)) {
+                  if (err.errno === error.ERRNO.INVALID_VERIFICATION_CODE && butil.buffersAreEqual(code, account.emailCode)) {
                     // The code is just for the account, not for any sessions
                     return true
-                  } else {
-                    throw error.invalidVerificationCode()
                   }
+                  throw err
                 })
                 .then(function () {
 
                   // If the account is already verified, they may be e.g.
                   // clicking a stale link.  Silently succeed.
                   if (account.emailVerified) {
-                    log.increment('account.already_verified')
+                    if (butil.buffersAreEqual(code, account.emailCode)) {
+                      log.increment('account.already_verified')
+                    }
```

Comment:
Without this condition, verification of session tokens would log spurious `account.already_verified` events.